File: adapters/utility-adapter/go/adapter/security_test.go
package adapter
import ("testing"; "time"; "fmt"; "crypto/hmac"; "crypto/sha256"; "encoding/hex")
func TestVerifySignature(t *testing.T) {
    secret := "test_secret"
    ts := fmt.Sprintf("%d", time.Now().Unix())
    body := []byte(`{"customer_ref":"123"}`)
    mac := hmac.New(sha256.New, []byte(secret))
    mac.Write([]byte(ts + ".")); mac.Write(body)
    sig := "v1=" + hex.EncodeToString(mac.Sum(nil))
    if !VerifySignature(secret, ts, sig, body, 300) { t.Error("Signature verification failed") }
    if VerifySignature(secret, ts, "v1=bad", body, 300) { t.Error("Should fail on bad signature") }
}

File: adapters/utility-adapter/nodejs/tests/security.test.ts

import * as crypto from 'crypto';
import { verifySignature } from '../src/security';

test('verifySignature returns true for valid signature', () => {
    const secret = 'test_secret';
    const ts = Math.floor(Date.now() / 1000).toString();
    const body = Buffer.from(JSON.stringify({ customer_ref: '123' }));
    const sig = 'v1=' + crypto.createHmac('sha256', secret).update(ts + '.' + body.toString()).digest('hex');
    expect(verifySignature(secret, ts, sig, body)).toBe(true);
});

test('verifySignature returns false for invalid signature', () => {
    const secret = 'test_secret';
    const ts = Math.floor(Date.now() / 1000).toString();
    const body = Buffer.from('{}');
    expect(verifySignature(secret, ts, 'v1=bad', body)).toBe(false);
});
